admin/query/src/main/java/org/codice/ditto/replication/admin/query/ReplicationUtils.java

```diff
@@ -88,20 +108,76 @@ public boolean siteIdExists(String id) {
     }
   }
 
-  public ReplicationSiteField updateSite(
-      String id, String name, AddressField address, @Nullable String rootContext) {
-    String urlString = addressFieldToUrlString(address, rootContext);
+  public boolean updateSite(
+      String id, @Nullable String name, AddressField address, @Nullable String rootContext) {
     ReplicationSite site = siteManager.get(id);
 
+    String updatedUrl = updateUrl(site.getUrl(), address, rootContext);
     setIfPresent(site::setName, name);
-    setIfPresent(site::setUrl, urlString);
+    setIfPresent(site::setUrl, updatedUrl);
+
+    try {
+      siteManager.save(site);
+    } catch (ReplicationPersistenceException e) {
+      LOGGER.debug("Unable to save updated site {} with id {}", site, id, e);
+      return false;
+    }
+
+    return true;
+  }
+
+  private String updateUrl(String prevUrl, AddressField addressField, @Nullable String context) {
+    URL url;
+    try {
+      url = new URL(prevUrl);
+    } catch (MalformedURLException e) {
+      // sites previous saved url should always be valid
+      return "";
+    }
+
+    final String oldHostname = url.getHost();
+    final int oldPort = url.getPort();
+    final String oldContext = stripStartingSlashes(url.getPath());
+
+    final String newHostname;
+    final int newPort;
+    final String newContext;
+
+    if (addressField.host().hostname() != null) {
+      newHostname = addressField.host().hostname();
+    } else {
+      newHostname = oldHostname;
+    }
+
+    if (addressField.host().port() != null) {
+      newPort = addressField.host().port();
+    } else {
+      newPort = oldPort;
+    }
+
+    if (context != null) {
+      newContext = stripStartingSlashes(context);
+    } else {
+      newContext = oldContext;
+    }
+
+    try {
+      return new URL(
+              String.format("%s://%s:%d/%s", url.getProtocol(), newHostname, newPort, newContext))
+          .toString();
+    } catch (MalformedURLException e) {
+      // all fields are validated by graphql before we make it here so this cannot be hit.
+      return "";
+    }
+  }
 
-    return getSiteFieldForSite(site);
+  private String stripStartingSlashes(String str) {
+    return str.trim().replaceFirst("^[/\\s]*", "");
```

Comment:
❓ Is the `trim()` really necessary here? I thought the regex was taking care of spaces too